Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -136,20 +136,21 @@ final class PurchasesOrchestrator {
             storeKit2StorefrontListener.listenForStorefrontChanges()
         }
 
-        Task {
-            #if os(iOS) || targetEnvironment(macCatalyst) || VISION_OS
-            #if swift(>=5.8)
-            if #available(iOS 16.4, *) {
+        #if os(iOS) || targetEnvironment(macCatalyst) || VISION_OS
+        if #available(iOS 16.4, *) {
+            Task {
```

Comment:
It's important that this happens in 2 separate `Tasks` so that this doesn't block `storeKit2TransactionListener`

Reply:
good catch